superset-frontend/src/views/CRUD/data/database/DatabaseModal/ExtraOptions.tsx

```diff
@@ -198,6 +198,23 @@ const ExtraOptions = ({
                 />
               </div>
             </StyledInputContainer>
+            <StyledInputContainer>
+              <div className="input-container">
+                <IndeterminateCheckbox
+                  id="allows_preview_data"
+                  indeterminate={false}
+                  checked={!!db?.extra_json?.allows_preview_data}
+                  onChange={onExtraInputChange}
+                  labelText={t('Allow SQL Lab data preview queries')}
+                />
+                <InfoTooltip
+                  tooltip={t(
+                    'The allows_preview_data field is a boolean specifying whether or not data ' +
+                      'preview queries will be run when fetching table metadata in SQL Lab.',
+                  )}
+                />
+              </div>
+            </StyledInputContainer>
```

Comment:
I believe we may need to reverse this flag to `disable_preview_data`, as otherwise this will disable data preview for existing databases (breaking change)